File: frontend/src/types/activity.ts
export type ActivityStatus =
  | 'DRAFT'
  | 'PENDING_REVIEW'
  | 'REJECTED'
  | 'APPROVED'
  | 'PUBLISHED'
  | 'ONGOING'
  | 'ENDED'
  | 'CANCELLED'

export interface Activity {
  id: number
  title: string
  description: string
  coverImage?: string
  organizerId: number
  location: string
  checkinLatitude?: number
  checkinLongitude?: number
  checkinRadiusMeters?: number
  locationCheckinRequired: boolean
  startTime: string
  endTime: string
  registrationStartTime: string
  registrationEndTime: string
  capacity: number
  currentRegisteredCount: number
  remainingCapacity: number
  status: ActivityStatus
  reviewComment?: string
  requireFeedback: boolean
  feedbackDeadline?: string
  createdAt: string
  updatedAt: string
}

export interface ActivityPayload {
  title: string
  description: string
  coverImage?: string
  location: string
  checkinLatitude?: number
  checkinLongitude?: number
  checkinRadiusMeters?: number
  startTime: string
  endTime: string
  registrationStartTime: string
  registrationEndTime: string
  capacity: number
  requireFeedback: boolean
  feedbackDeadline?: string
}

File: backend/src/main/java/com/example/activity/entity/Activity.java
package com.example.activity.entity;

import com.baomidou.mybatisplus.annotation.IdType;
import com.baomidou.mybatisplus.annotation.TableId;
import com.baomidou.mybatisplus.annotation.TableName;
import lombok.Data;

import java.time.LocalDateTime;

@Data
@TableName("activities")
public class Activity {

    @TableId(type = IdType.AUTO)
    private Long id;
    private String title;
    private String description;
    private String coverImage;
    private Long organizerId;
    private String location;
    private Double checkinLatitude;
    private Double checkinLongitude;
    private Integer checkinRadiusMeters;
    private LocalDateTime startTime;
    private LocalDateTime endTime;
    private LocalDateTime registrationStartTime;
    private LocalDateTime registrationEndTime;
    private Integer capacity;
    private Integer currentRegisteredCount;
    private ActivityStatus status;
    private String reviewComment;
    private Boolean requireFeedback;
    private LocalDateTime feedbackDeadline;
    private LocalDateTime createdAt;
    private LocalDateTime updatedAt;
}
